src/streaming-client.ts

```diff
@@ -145,20 +152,21 @@ export class RevAiStreamingClient extends EventEmitter {
                     }
                 }
             });
-            this.doSendLoop(connection, this.requests);
+            this.doSendLoop(conn, this.requests);
         });
     }
 
-    private doSendLoop(connection: any, buffer: PassThrough): void {
-        if (connection.connected) {
-            let value = buffer.read(buffer.readableLength);
+    private doSendLoop(conn: connection, buffer: PassThrough): void {
+        if (conn.connected) {
+            const value = buffer.read();
             if (value !== null) {
-                connection.send(value);
-                if (value.includes('EOS') || value.includes(Buffer.from('EOS'))) {
-                    connection.sendUTF('EOS');
+                if (typeof value === 'string' && value === 'EOS') {
```

Comment:
Should we simplify this? It could be
`if (typeof value === 'string') { conn.sendUTF(value) }`

This would make the SDK more flexible but would make it so customers can mess up the protocol more easily

Reply:
Yeah I'm fine with this